New testsuite failures with LinearOperator

@tamiko : The 'tester' fails with errors like this:
```
FAILED: /usr/bin/c++   -DDEBUG -DSOURCE_DIR=\"/node/bangerth/trunk/dealii/tests/lac\" -I/node/bangerth/trunk/build/include -I/node/bangerth/trunk/dealii/include -I/node/bangerth/trunk/dealii/bundled/umfpack/UMFPACK/Include -I/node/bangerth/trunk/dealii/bundled/umfpack/AMD/Include -I/node/bangerth/trunk/dealii/bundled/boost-1.56.0/include -I/node/bangerth/trunk/dealii/bundled/muparser_v2_2_3/include -I/apps/openmpi/1.6.5/include -I/node/bangerth/bin/trilinos-11.6.1/include -I/node/bangerth/bin/petsc-3.4.4/include -I/node/bangerth/bin/petsc-3.4.4/x86_64/include -I/node/bangerth/bin/p4est-0.3.4.2/FAST/include    -pedantic -fpic -Wall -Wextra -Wpointer-arith -Wwrite-strings -Wsynth -Wsign-compare -Wswitch -Wno-long-long  -std=c++0x -O0 -ggdb -Wa,--compress-debug-sections -MMD -MT CMakeFiles/linear_operator_04.debug.dir/linear_operator_04.cc.o -MF "CMakeFiles/linear_operator_04.debug.dir/linear_operator_04.cc.o.d" -o CMakeFiles/linear_operator_04.debug.dir/linear_operator_04.cc.o -c /node/bangerth/trunk/dealii/tests/lac/linear_operator_04.cc
In file included from /node/bangerth/trunk/dealii/tests/lac/linear_operator_04.cc:22:0:
/node/bangerth/trunk/dealii/include/deal.II/lac/linear_operator.h: In function ‘dealii::LinearOperator<Range, Domain> dealii::linear_operator(const OperatorExemplar&, const Matrix&)’:
/node/bangerth/trunk/dealii/include/deal.II/lac/linear_operator.h:1189:54: error: no default argument for ‘OperatorExemplar’
/node/bangerth/trunk/dealii/include/deal.II/lac/linear_operator.h:1189:54: error: no default argument for ‘Matrix’
```
See, for example, http://cdash.kyomu.43-1.org/testDetails.php?test=10643725&build=6017

I have no idea what this error means, to be honest.

There are other errors as well:
http://cdash.kyomu.43-1.org/testDetails.php?test=10643729&build=6017

For reference, this is with GCC 4.6.3. It may simply still have bugs in handling lambda functions.

--- a/include/deal.II/lac/linear_operator.h
+++ b/include/deal.II/lac/linear_operator.h
@@ -1144,9 +1144,7 @@ namespace
  *
  * @ingroup LAOperators
  */
-template <typename Range = Vector<double>,
-          typename Domain = Range,
-          typename Matrix>
+template <typename Range, typename Domain, typename Matrix>
 LinearOperator<Range, Domain> linear_operator(const Matrix &matrix)
 {
   // implement with the more generic variant below...
@@ -1171,13 +1169,12 @@ LinearOperator<Range, Domain> linear_operator(const Matrix &matrix)
  *
  * @ingroup LAOperators
  */
-template <typename Range = Vector<double>,
-          typename Domain = Range,
+template <typename Range,
+          typename Domain,
           typename OperatorExemplar,
           typename Matrix>
 LinearOperator<Range, Domain>
-linear_operator(const OperatorExemplar &operator_exemplar,
-                const Matrix &matrix)
+linear_operator(const OperatorExemplar &operator_exemplar, const Matrix &matrix)
 {
   LinearOperator<Range, Domain> return_op;
 

